slack-lint-checks/src/main/java/slack/lint/rx/RxObservableEmitDetector.kt

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2025 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package slack.lint.rx
+
+import com.android.tools.lint.checks.DataFlowAnalyzer
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.uast.UCallExpression
+import org.jetbrains.uast.ULambdaExpression
+import slack.lint.util.sourceImplementation
+
+class RxObservableEmitDetector : Detector(), SourceCodeScanner {
+  override fun getApplicableUastTypes() = listOf(UCallExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+      override fun visitCallExpression(node: UCallExpression) {
+        val issue = functionToIssue[node.methodName] ?: return
+        val lambdaExpression = node.valueArguments.lastOrNull() as? ULambdaExpression ?: return
+        val producerScopeParam = lambdaExpression.parameters.firstOrNull() ?: return
+
+        // Verify the parameter is a ProducerScope
+        if (!producerScopeParam.type.canonicalText.startsWith(PROVIDER_SCOPE_FQN)) return
+
+        var sendCalled = false
+
+        val visitor =
+          object : DataFlowAnalyzer(emptySet()) {
+            override fun visitCallExpression(node: UCallExpression): Boolean {
+              // If we find a nested factory method, return immediately and stop traversing this
+              // code path.
+              // Note: this factory will be validated by the UElementHandler above
+              if (node.methodName in functionToIssue) return true
+
+              return super.visitCallExpression(node).also {
+                if (node.hasProviderScopeReceiver() && node.methodName in REQUIRE_ONE_OF)
+                  sendCalled = true
+              }
```

Comment:
nit: this `also` doesn't actually use the `it` argument and looks a bit out of place, let's just move it to its own statement above?

Reply:
yeah I realize that. I have been unable to actually find `it`. I suspect I'm missing something here (or maybe made a mistake) but I have hit a wall. Any suggestions would be most welcome

Reply:
yup, can do. I thought somewhere you had suggested that calling `super.visitCallExpression` should be the first thing this method does. I don't see that so must have imagined it. I'll update

Reply:
https://github.com/slackhq/slack-lints/pull/375/commits/ebd0d55e30159243898c17503a837e25121d6587